dataset/include/scipp/dataset/groupby.h

```diff
@@ -17,7 +17,7 @@ namespace scipp::dataset {
 class SCIPP_DATASET_EXPORT GroupByGrouping {
 public:
   using group = boost::container::small_vector<Slice, 4>;
-  GroupByGrouping(Variable &&key, std::vector<group> &&groups)
+  GroupByGrouping(Variable key, std::vector<group> groups)
```

Comment:
Why do we no longer pass by reference here?

Reply:
To cover both cases: Passing a temporary (moved) as well as making a copy, see https://github.com/scipp/scipp/pull/1359#discussion_r516647717.